### DISCUSSION TITLE
Is the "Modify the LAScatalog drivers" page still valid with locate_trees

### DISCUSSION THREAD

**Discussion by djes** [r-lidar/lidR#616] [Q&A] [answered]
Hello,

As I'm having problems exporting to shapefiles due to the bitmerge value overflow, I'm trying to export `locate_trees` output to a single file (sqlite or gpkg) but I encounter a difficulty. The "Modify the LAScatalog drivers" page talks about `SpatialPointsDataFrame` and uses find_trees, but the `locate_trees` documentation says that the output is `sf`.  

So I've rewritten the functions :

```r
# Create a function that writes a sf into
# a SQLite database.
dbWrite_sf = function(x, path, name)
{
  x <- as.data.frame(x)
  con <- RSQLite::dbConnect(RSQLite::SQLite(), path)
  RSQLite::dbWriteTable(con, name, x, append = TRUE)
  RSQLite::dbDisconnect(con)
}

# Change the driver used to write sf objects
# User-defined drivers have the precedence

ctg_filter@output_options$drivers$sf = list(
   write = dbWrite_sf,
   extension = ".sqlite",
   object = "x",
   path = "path",
   param = list(name = "trees"))

opt_output_files(ctg_filter) <- paste0("tops/", "trees")
```

But I'm not sure about the `x <- as.data.frame(x)`

Is it the correct way to handle this ? Thank you...

**Comment by Jean-Romain** [comment, marked as answer]
The wiki page is outdated but the general idea did not change. For `sf` `st_write` already provide the internals to write in databases. I don't think you need to use bare bones `RSQLite` functions.

`as.data.frame` on `Spatial*DataFrame` returns the coordinates + the attributes. This is not the case for `sf` where it returns a `data.frame` with a `sfc` column that won't be writable.

**Reply by djes** [reply]
Ok, thank you, I'll check this !
